removes boost::asio:: from example in doc

tnixeu opened the pull request:
In an example `boost::asio::io_service ios;` is used like this and looks in the generated doc like `boost::asio::boost::asio::io_service ios;` This pull request removes `boost::asio::` from the example.

tnixeu commented:
I fixed some more of those issues. I am not sure if I have to change the pull request or make a new one. I see both commits listed here. So I think I do not have to make a new pull request.

klemens-morgenstern commented:
Why would you remove `boost::asio`? I mean I should rename it to `io_context`, but I don't see why you would remove the scope.

keith-bennett-gbg commented:
There's a few places where the namespace prefix is duplicated such that it says `boost::asio::boost::asio::`in the rendered documentation. Which is fine enough for someone who's paying attention and knows that typos happen and can reason about it. But it's annoying anyway. I think that's partially what this is trying to resolve.

tnixeu commented:
I removed it, because it is added during generating the doc again. That is the first commit.
In the second commit I fixed also some other replacement issues. Please have a look at the generated doc.

klemens-morgenstern commented:
Oh, right, because the macro generates the prefix. Thanks, I'll update it to 'io_context' in another place.
